catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -126,7 +136,30 @@ private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
                 filterBuilder);
 
         Stopwatch stopwatch = Stopwatch.createStarted();
-        QueryResponse response = catalogFramework.query(request);
+        QueryResponse response = null;
+
+        if (configuration.isDisableLocalCatalog()) {
+            Set<String> sourceIds;
+            if (CollectionUtils.isNotEmpty(request.getSourceIds())) {
+                sourceIds = new HashSet<>(request.getSourceIds());
+            } else {
+                sourceIds = new HashSet<>(catalogFramework.getSourceIds());
+            }
+            sourceIds.remove(catalogFramework.getId());
+            request = new QueryRequestImpl(request.getQuery(),
+                    false,
+                    sourceIds,
+                    request.getProperties());
+
+            if (CollectionUtils.isNotEmpty(sourceIds)) {
```

Comment:
Per the API documentation for Federatable, https://github.com/troymohl/ddf/blob/bb7177ea3dd800f5d267f8bf41070e704fc9b2bb/catalog/core/catalog-core-api/src/main/java/ddf/catalog/federation/Federatable.java#L45
Setting enterprise to true is supposed to query all sources, so the intent of this code was to modify the query and remove the local source id from the list and then do a query to the other sources. In the case that a source is provided that is the local source (which shouldn't be possible since the UI doesn't show it), then this code would return an empty response. Otherwise, it performs the query as requested.